Upgrade to NXP Release 5.4.24-2.1.0 (Part 2)

zandrey opened the pull request:
This is a Part 2 of the upgrade to NXP Release 5.4.24-2.1.0 (referred to in #456). It mainly addresses `firmware`, `optee` and various multimedia and graphics recipes.

@panfir Commit 022158c in this PR covers the PR #396, and it supersedes it. After the integration, PR #396 can be closed.

@thochstein Commit 145a2c4 has a patch for `imx-gst1.0-plugin` which solves the compilation issue with GCC10 and `-fno-common`, you might want to pick it up internally.

This upgrade has been tested on *imx8mmevk*, Gstreamer Video playback is operable.

-- andrey

otavio commented on recipes-security/optee-imx/optee-os/0001-optee-os-fix-gcc10-compilation-issue-and-missing-cc-.patch:
```diff
@@ -89,7 +93,7 @@ index c05815f3..68f45552 100644
  PLATFORM_$(PLATFORM) := y
 @@ -16,10 +17,6 @@ PLATFORM_FLAVOR_$(PLATFORM_FLAVOR) := y
- $(call cfg-depends-all,CFG_PAGED_USER_TA,CFG_WITH_PAGER CFG_WITH_USER_TA)
+ $(eval $(call cfg-depends-all,CFG_PAGED_USER_TA,CFG_WITH_PAGER CFG_WITH_USER_TA))
```
Why use this `eval` here? Doesn't the old `call` work?

zandrey replied:
This is done in the upstream with commit [6b385aaf7a1e83b689e90a4c4290d5406b172ca2](https://github.com/jforissier/optee_os/commit/6b385aaf7a1e83b689e90a4c4290d5406b172ca2)

NXP updated its fork, so that commit got integrated as well.

What I did in here is I re-based the patch to the latest version in the fork, hence you see a bit more differences here. 😉

otavio replied:
Ol, cool.

otavio commented on recipes-multimedia/imx-vpuwrap/imx-vpuwrap_4.5.5.bb:
```diff
@@ -0,0 +1,29 @@
+# Copyright (C) 2013-2016 Freescale Semiconductor
```
@thochstein could you find out the logs?

thochstein replied:
```
r60874@tx30imx-01:~/git/vpu_wrapper$ git log --oneline --no-decorate 8a3f1b0..b683295
b683295 MA-17072 [VC8000E] Set AVC default profile/level
ae24b72 MA-16984 update Android.bp to remove "imx_defaults"
1349ea2 MA-16873-4 Update vpu_wrapper path
c1235d2 MA-16830-2 Fix memleak when free memory in wrapper
f680b3c MMFMWK-8745 hantro: enable reorder config for hantro vpu
3685f06 MA-16717-3 add VPU_DEC_SKIP for CODEC_PIC_SKIPPED
35b43ea MA-16717-2 Add Config to reset codecdata flag
8c92057 MMFMWK-8742 Map decoders to timestamps
ad0bc7f MMFMWK-8739 [8mp]Grecorder: Only one recorded ts file play well when set '-n 3'
6c306bc MMFMWK-8738 [8mp] Set right input buffer's Y,U,V physical address
1524207 MMFMWK-8736 Fix hantro decoder can't get fullrange flag of files encoded with vc8000e
19fe757 MA-16593 [evk_8mm] Fix video hang due to error frame
13459f7 MMFMWK-8733 Update EULA.txt for vpu_wrapper according to imx-release-info
35ea5e6 [configurea.ac] Update version to 4.5.5 for next release
94c38de MA-16435-1 Modify Android makefile
96e189f MMFMWK-8720 Replace docs/EULA.txt and add SRC file SCR-imx-vpuwrap.txt
341dc9b MMFMWK-8720 Simplify vpu wrapper
bf6680e MMFMWK-8720 Sync vpu wrapper code with Android and change copyright
1e6bc61 MMFMWK-8720 fix vc8000e encoder can't work
a61d115 MMFMWK-8722 Fix color abnormal issue of streaming feature
f68972d MMFMWK-8722 [8mp] Fix bitrate getting smaller issue of VC8000E encoder
e7b3992 MMFMWK-8718 Enable cropping for hantro encoder at preprocess stage
017ecf7 MMFMWK-8714 [8mp] Adjust parameters to get correct filesize with given bit rate.
95acceb MMFMWK-8595 seperate 845 and 865 rootfs
180d19f MMFMWK-8595 Enable VC8000E encoder for 865
6fb749b [configurea.ac] Update version to 4.5.4 for next release
```

otavio commented on recipes-multimedia/imx-codec/imx-codec_4.5.5.bb:
```diff
@@ -5,14 +5,14 @@
 LICENSE = "Proprietary"
 SECTION = "multimedia"
-LIC_FILES_CHKSUM = "file://COPYING;md5=fd4b227530cd88a82af6a5982cfb724d"
+LIC_FILES_CHKSUM = "file://COPYING;md5=228c72f2a91452b8a03c4cab30f30ef9"
```
@thochstein could you find the logs?

thochstein replied:
```
$ git log --oneline --no-decorate 4ebbb1bc..16dba069
16dba069 MMFMWK-8757 release/build_std: Fix yocto automation issue
b912378e MMFMWK-8755 release/build_std: Add flac unit test to pkg
5871d696 MLK-23360-2: release/lib: Update mp3 codec lib
3573ebeb MLK-23360-1: mp3_dec: MP3 codec doesn't detect sample rate change in the stream
edb24319 [configurea.ac] Update dsp version to 1.0.14 for next release
980baaad [configurea.ac] Update version to 4.5.5 for next release
4fddfbd5 [configurea.ac] Update dsp version to 1.0.13 for next release
e3b603f8 [configurea.ac] Update version to 4.5.4 for next release
```

otavio commented on recipes-multimedia/imx-parser/imx-parser_4.5.5.bb:
```diff
@@ -1,11 +1,11 @@
 # Copyright (C) 2012-2018 O.S. Systems Software LTDA.
 # Copyright (C) 2012-2016 Freescale Semiconductor
-# Copyright 2017, 2019 NXP
+# Copyright (C) 2017, 2019 NXP
```
@thochstein could you find the logs?

thochstein replied:
```
~/git/fsl_mad_multimedia_layer$ git log --oneline --no-decorate eff47177..72395663
72395663 MA-17095: MP4 PARSER 06.16.07
6abeb330 MA-16391-2: MP4 PARSER 06.16.06
7a8c66d5 MA-16391: MP4 PARSER 06.16.05 support ENCRYPTED_BYTE_BLOCK, SKIP_BYTE_BLOCK, CRYPTO_IV
6f7d785f MA-17070 MP3 PARSER 03.03.03
8fb0fc1a MA-17026 AAC PARSER 03.01.04
8d04d6cd MA-16982: fix cts android.media.cts.DecoderTest#testDecodeMonoOggMkv
1bc9b758 MA-16969 [Android] Q10_2.0.0 cts MediaMuxerTest#testAllTimestampsBVideoOnly fail
efa3c12f MMFMWK-8754 MPEG2 PARSER 05.01.02
fbab3462 MA-16828: fix cts android.media.cts.DecoderTest#testTrackSelectionMkv
2e2ca2ad MA-16684: suppoert av1 video format in mkv parser
e9cf3560 [configurea.ac] Update version to 4.5.5 for next release
138ef113 MA-16403 Fix ID3 parser return wrong artwork size
782e91e1 MA-16375 MPEG2 PARSER 05.01.01
57207e77 MMFMWK-8725 MP3 PARSER 03.03.01
48581949 MMFMWK-8650 MPG2 PARSER 04.06.03
2f3da775 [configurea.ac] Update version to 4.5.4 for next release
```

zandrey commented on recipes-bsp/firmware-imx/files/epdc:
```diff
@@ -0,0 +1,12 @@
+	echo 1 > loading
+	cat /lib/firmware/imx/epdc/epdc_ED060XH2C1.fw > data
+	echo 0 > loading
+fi
```
@otavio Agree. 👍 

This whole `firmware-imx` was indeed a controversial update, but I decided to include it anyway - mainly to get a feedback regarding the implementation.

I would split this commit and drop the *systemd* part for it to be re-worked.

otavio replied:
Cool; this allow us to go ahead with this and rework the load next.

panfir commented:
> This is a Part 2 of the upgrade to NXP Release 5.4.24-2.1.0 (referred to in #456). It mainly addresses `firmware`, `optee` and various multimedia and graphics recipes.
> 
> @panfir Commit [022158c](https://github.com/Freescale/meta-freescale/commit/022158ce528fec1152d75670e0a04175956c9ea8) in this PR covers the PR #396, and it supersedes it. After the integration, PR #396 can be closed.
> 
> @thochstein Commit [145a2c4](https://github.com/Freescale/meta-freescale/commit/145a2c424d3bd580d78a44e7ac9cabf00ac5dcbc) has a patch for `imx-gst1.0-plugin` which solves the compilation issue with GCC10 and `-fno-common`, you might want to pick it up internally.
> 
> This upgrade has been tested on _imx8mmevk_, Gstreamer Video playback is operable.
> 
> -- andrey

@zandrey Sure, go ahead, thanks
@otavio, there were any particular reason PR #396 was not accepted some months ago?

otavio commented:
> @otavio, there were any particular reason PR #396 was not accepted some months ago?

I think it ended getting lost. I don't recall if we found any issues at last review or not but we should improve this in the future.

It'd be nice if you could review it here once we address the issues.

panfir commented:
> > @otavio, there were any particular reason PR #396 was not accepted some months ago?
> 
> I think it ended getting lost. I don't recall if we found any issues at last review or not but we should improve this in the future.
> 
> It'd be nice if you could review it here once we address the issues.

The main issue there was the systemd services, so for better covering, I implemented the udev rules, as you recommended.

I remember I tested in labgrid on various platforms. The testing team which I worked with at that time, also confirmed the firmware loading, since not all firmware could be easily checked via dmesg, most probably is about how was written some of drivers.

otavio commented:
@panfir may I ask you to update it to the latest release and prepare a new PR? so @zandrey could remove it from this PR and we handle it individually.

zandrey commented:
> The main issue there was the systemd services, so for better covering, I implemented the udev rules, as you recommended.

@panfir Is there any reason that implementation was not made a part of `5.4.24-2.1.0` release? Recipe versions targeted in PR #396 are `8.6` while `5.4.24-2.1.0` has `8.8` and udev rules were not part of it... 😕

panfir commented:
> > The main issue there was the systemd services, so for better covering, I implemented the udev rules, as you recommended.
> 
> @panfir Is there any reason that implementation was not made a part of `5.4.24-2.1.0` release? Recipe versions targeted in PR #396 are `8.6` while `5.4.24-2.1.0` has `8.8` and udev rules were not part of it...

There was no reason, udev implementation was first done upstream. Usually after upstream acceptance we consider an alignment of the nxp code. That was not the case here. On the other hand internally we do not have a problem with the implementation of systemd services.

panfir commented:
> @panfir may I ask you to update it to the latest release and prepare a new PR? so @zandrey could remove it from this PR and we handle it individually.

Not really, it would mean redoing the setup and all that validation work, it would take some time. Andrey is already working on this and probably closer to the end.

otavio commented:
@panfir ok; in any case you'll need to redo the tests as we don't have all the setup for it. Thanks for your support.

zandrey commented:
> There was no reason, udev implementation was first done upstream.

@panfir Would it be possible to get those? I cannot seem to find it anywhere...

I guess if you've done some work here - I can just take it over rather than re-doing it again and then having troubles of syncing 2 implementations once you would have them in the next BSP.

Can you please tell me where those could be obtained?

Thanks a lot!

-- andrey

panfir commented:
> > There was no reason, udev implementation was first done upstream.
> 
> @panfir Would it be possible to get those? I cannot seem to find it anywhere...
> 
> I guess if you've done some work here - I can just take it over rather than re-doing it again and then having troubles of syncing 2 implementations once you would have them in the next BSP.
> 
> Can you please tell me where those could be obtained?
> 
> Thanks a lot!
> 
> -- andrey

I meant, I implemented first 'for upstream' ..in that pull request. I don't think is anywhere upstream. Sorry for confusion.

zandrey commented:
> 
> 
> > > There was no reason, udev implementation was first done upstream.
> > 
> > 
> > @panfir Would it be possible to get those? I cannot seem to find it anywhere...
> > I guess if you've done some work here - I can just take it over rather than re-doing it again and then having troubles of syncing 2 implementations once you would have them in the next BSP.
> > Can you please tell me where those could be obtained?
> > Thanks a lot!
> > -- andrey
> 
> I meant, I implemented first 'for upstream' ..in that pull request. I don't think is anywhere upstream. Sorry for confusion.

OK, got it! Thanks for clarifications, I'd take this one further then.

-- andrey

zandrey commented:
@otavio I've updated the `firmware-imx` recipe in commit and dropped all scripts and systemd units for now, we can deal with them later.

@panfir Can you please have a look at `filrmware-imx` changes and let me know if that is OK or you spot something I missed out?

@thochstein Would it be possible to get those change  logs @otavio asked above? I'd be waiting for them to include this info in commit message and then re-base the branch again (hopefully for the last time 😉). Thanks a lot!

-- andrey

otavio commented on recipes-bsp/firmware-imx/firmware-imx_8.8.bb:
```diff
@@ -0,0 +1,109 @@
+PACKAGES =+ "${PN}-epdc ${PN}-sdma ${PN}-easrc ${PN}-hdmi ${PN}-xcvr ${PN}-xuvi"
+
+FILES_${PN}-epdc = "${nonarch_base_libdir}/firmware/imx/epdc/"
+FILES_${PN}-sdma = "${nonarch_base_libdir}/firmware/imx/sdma/sdma-imx6q.bin ${nonarch_base_libdir}/firmware/imx/sdma/sdma-imx7d.bin"
```
This is intended to be split in individual packages on line 89, and not included on this package.

zandrey replied:
True, it has slipped my mind. It is actually required to be separate via `MACHINE_FIRMWARE` appends.

Would put it into separate packages then.

zandrey commented on recipes-bsp/firmware-imx/firmware-imx_8.8.bb:
```diff
@@ -0,0 +1,109 @@
+FILES_${PN}-easrc = "${nonarch_base_libdir}/firmware/imx/easrc/"
+FILES_${PN}-hdmi = "${nonarch_base_libdir}/firmware/imx/hdmi/"
+FILES_${PN}-xcvr = "${nonarch_base_libdir}/firmware/imx/xcvr/"
+FILES_${PN}-xuvi = "${nonarch_base_libdir}/firmware/imx/xuvi/"
```
I guess `easrc`, `xcrv` and (potentially) `xuvi` would make sense to pass through the split function.

`hdmi` and `epdc` are a bit special though, as they do not have a dedicated files pattern to form package names. I'm afraid if we go further with the split for those - we would need to add more logic and granularity in `do_install()` then.

If you're OK with that - I would go on with first three, leaving `hdmi` and `epdc` as-is.

otavio replied:
Sure, let's do baby steps so we can improve it along the way.

zandrey commented:
@otavio Done, `firmware-imx` recipe is re-worked. Please have a look at it as I changes a bit the logic regarding the main package, which stays empty by intention. I've explained that in the comments in recipe so people are not getting confused.

-- andrey

thochstein commented:
> @thochstein Would it be possible to get those change logs @otavio asked above? I'd be waiting for them to include this info in commit message and then re-base the branch again (hopefully for the last time 😉). Thanks a lot!

Sorry for the delay, I'll work on those changelogs now.

zandrey commented:
@thochstein Thanks for the update regarding commit logs! I've updated commit descriptions to include those logs in.

-- andrey
